pkg/config/config.go

```diff
@@ -422,35 +421,21 @@ func (c *MicroshiftConfig) ReadFromConfigFile(configFile string) error {
 		c.KASAdvertiseAddress = config.ApiServer.AdvertiseAddress
 	}
 
-	if config.Etcd.DefragCheckFreq != "" {
-		d, err := time.ParseDuration(config.Etcd.DefragCheckFreq)
+	if config.Etcd.MemoryLimit != "" {
+		min, err := resource.ParseQuantity(EtcdMinimumMemoryLimit)
 		if err != nil {
-			return fmt.Errorf("failed to parse etcd defragCheckFreq: %v", err)
+			return fmt.Errorf("failed to parse etcd memory limit: %v", err)
 		}
-		c.Etcd.DefragCheckFreq = d
-	}
-	if config.Etcd.MinDefragSize != "" {
-		q, err := resource.ParseQuantity(config.Etcd.MinDefragSize)
-		if err != nil {
-			return fmt.Errorf("failed to parse etcd minDefragSize: %v", err)
-		}
-		if !q.IsZero() {
-			c.Etcd.MinDefragBytes = q.Value()
-		}
-	}
-	if config.Etcd.MaxFragmentedPercentage > 0 {
-		c.Etcd.MaxFragmentedPercentage = config.Etcd.MaxFragmentedPercentage
-	}
-	if config.Etcd.QuotaBackendSize != "" {
-		q, err := resource.ParseQuantity(config.Etcd.QuotaBackendSize)
+
+		limit, err := resource.ParseQuantity(config.Etcd.MemoryLimit)
 		if err != nil {
-			return fmt.Errorf("failed to parse etcd quotaBackendSize: %v", err)
+			return fmt.Errorf("failed to parse etcd memory limit: %v", err)
 		}
-		if !q.IsZero() {
-			c.Etcd.QuotaBackendBytes = q.Value()
+		if limit.Value() < min.Value() {
+			return errors.New("attempted to set etcd memory limit beneath the minimum of 50MB, please select a higher value.")
```

Comment:
This error message should use the default value from the variable instead of repeating it.

Reply:
So you mean to say that if limit is less than min, use the min value silently?